src/charm.py

```diff
@@ -139,10 +122,33 @@ def reconcile(self, _: ops.HookEvent) -> None:
             self.unit.name,
             self.state.cluster_key,
         )
+
+    def reconcile(self, _: ops.HookEvent) -> None:
+        """Reconcile Wazuh configuration with charm state.
+
+        This is the main entry for changes that require a restart.
+        """
+        container = self.unit.get_container(wazuh.CONTAINER_NAME)
+        if not container.can_connect():
+            logger.warning(
+                "Unable to connect to container during reconcile. "
+                "Waiting for future events which will trigger another reconcile."
+            )
+            self.unit.status = ops.WaitingStatus("Waiting for pebble.")
+            return
+        if not self.state:
+            self.unit.status = ops.WaitingStatus("Waiting for status to be available.")
+            return
+        self._configure_installation()
+
         container.add_layer("wazuh", self._wazuh_pebble_layer, combine=True)
+        # The prometheus exporter requires the users to be set up
+        logger.debug("Unconfigured API users %s", self.state.unconfigured_api_users)
+        if not self.state.unconfigured_api_users:
+            logger.debug("Adding prometheus pebble layer")
+            container.add_layer("prometheus", self._prometheus_pebble_layer, combine=True)
```

Comment:
We're adding the prometheus layer in this condition and in the other if statement which is the opposite of this condition. Shouldn't we add the layer systematically before setting the workload version and replan there ?

Reply:
Indeed. This was a leftover from some other approach I was testing